shell/platform/android/io/flutter/view/FlutterMain.java

```diff
@@ -56,6 +56,13 @@
     private static final String DEFAULT_KERNEL_BLOB = "kernel_blob.bin";
     private static final String DEFAULT_FLUTTER_ASSETS_DIR = "flutter_assets";
 
+    private static boolean isRunningInRobolectricTest = false;
```

Comment:
We spoke offline and I attempted this switch. It looks like we get into a quagmire if we try to control things at the `FlutterJNI` level instead of `FlutterMain`. We will begrudgingly accept this flag for now with a goal to improve this mechanism in the future.